src/app/Http/Controllers/SearchController.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Post;
+use App\Models\User;
+use App\Http\Requests\SearchRequest;
+use Illuminate\Contracts\View\View;
+
+class SearchController extends Controller
+{
+    public function search(SearchRequest $request): View
+    {
+        $query = $request->input('query');
+
+        // Perform the search query on the User model
+        $userResults = User::where('name', 'like', "%$query%")->get();
+
+        // Perform the search query on the Post model
+        $postResults = Post::where('content', 'like', "%$query%")->get();
+
+        return view('search.results', compact('userResults', 'postResults'));
```

Comment:
Why this is being exposed in the controller? You don't have service layer for this??